src/app/jscaip/MoveCoord.ts

```diff
@@ -1,20 +1,16 @@
 import { Move } from './Move';
 import { Coord } from './Coord';
-import { MoveEncoder } from '../utils/Encoder';
+import { Encoder } from '../utils/Encoder';
 import { MGPFallible } from '../utils/MGPFallible';
 
 export abstract class MoveCoord extends Move {
 
-    public static getEncoder<T extends MoveCoord>(generate: (coord: Coord) => T): MoveEncoder<T> {
-        return MoveEncoder.tuple(
+    public static getEncoder<T extends MoveCoord>(generate: (coord: Coord) => MGPFallible<T>): Encoder<T> {
+        return Encoder.tuple(
             [Coord.encoder],
             (m: T): [Coord] => [m.coord],
-            (fields: [Coord]): T => generate(fields[0]));
+            (fields: [Coord]): T => generate(fields[0]).get());
     }
-    public static getFallibleEncoder<T extends MoveCoord>(generate: (coord: Coord) => MGPFallible<T>): MoveEncoder<T> {
```

Comment:
on perd la possibilité d'avoir facilement un encoder de .of :/
Si le .from est plus fréquent, je serais pour avoir getNonFallibleEncoder et getEncoder

Reply:
Moins de code c'était pas mieux ?
ça et l'uniformité ?

Reply:
c'est plus de code à cet endroit pour moins de code à chaque utilisation sur du .of, donc moins de code au total en gardant les deux.
Puis vu qu'on a of *et* from, l'uniformité joue pas ici non ?